ieeeuoft/newhacks-site-2024
# Initial Registration Fields Setup

## Overview

TBD


## Progress

- Need to double check if error with copy-pasting file changes 
- At the moment, getting this error when doing `python manage.py migrate`

## Debugging Notes (for reference, will add to documentation later)

If you see the following issue:
<img width="973" alt="Screenshot 2024-06-28 at 12 35 52 AM" src="https://github.com/ieeeuoft/newhacks-site-2024/assets/80921817/1acb0e48-7d03-43f7-ac41-20bd3f9a522e">

Solution is to:

- Make sure you are running the correct Docker containers 
     - Specifically, you need to modify the file: `development/docker-compose.yml` to ensure the docker container have the correct name 
     - Rerun the docker containers (ie: do `docker-compose -f development/docker-compose.yml up -d`)
     - Since you've created new docker containers, you would need to create a new super-user
           - To do that, you need to enter the command: `python manage.py createsuperuser`
     - Do `python manage.py makemigrations` followed by `python manage.py migrate`

## Future changes to be made (after debugging) 

- In `hacathon_site/event/serializers.py`, need to update participant package to 2024

## Other Notes

- Main reference PR from last year: https://github.com/ieeeuoft/newhacks-site-2023/pull/6
- Other PR to reference:
     - https://github.com/ieeeuoft/makeuoft-site-2024/pull/7 (Updating Application Fields) 
          - Most important fields are: T-shirt size, Dietary Restrictions 
     - https://github.com/ieeeuoft/makeuoft-site-2024/pull/10 (Fixing some registration bugs. Can skip the YT Promo video) 
     - In general, reference the `makeuoft2024 repo` for most up-to-date form registration information


## Steps to QA

- Make sure that changes from the very 1st commit `re-add  qrcode dependencies` do not actually break deployment. This is the only non-registration related commit
